app/src/main/java/com/bashkevich/androidfundamentals/movieslist/viewmodel/MoviesListViewModel.kt

```diff
@@ -1,29 +1,32 @@
 package com.bashkevich.androidfundamentals.movieslist.viewmodel
 
 import android.util.Log
-import androidx.lifecycle.LiveData
-import androidx.lifecycle.MutableLiveData
-import androidx.lifecycle.ViewModel
-import androidx.lifecycle.viewModelScope
-import com.bashkevich.androidfundamentals.model.MoviesRepository
+import androidx.lifecycle.*
+import com.bashkevich.androidfundamentals.model.repository.MoviesRepository
 import com.bashkevich.androidfundamentals.model.viewobject.Movie
-import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
-import kotlinx.coroutines.withContext
 
 class MoviesListViewModel(private val moviesRepository: MoviesRepository) : ViewModel() {
+    private val _moviesListLiveData = MutableLiveData<List<Movie>>()
 
-    val moviesListLiveData: LiveData<List<Movie>> = moviesRepository.getAllMovies()
+    val moviesListLiveData: LiveData<List<Movie>>
+        get() = _moviesListLiveData
+
+    private val _errorLiveData = MutableLiveData<String>()
+
+    val errorLiveData: LiveData<String>
+        get() = _errorLiveData
 
     fun loadMoviesList() {
         viewModelScope.launch {
             try {
-                moviesRepository.refreshMovies()
+                _moviesListLiveData.value = moviesRepository.getAllMovies().asLiveData().value
```

Comment:
What does return `getAllMovies()`? Is it LiveData, just list or Flow?

If it is live data - can you use Transformations.map on it but just pass data to the live data that is subscribed in UI?

Reply:
GetAllMovies returns a Flow, so I transform it to LiveData and then subscribe to it

Reply:
Sorry, you don't subscribe to the flow. You convert it to live data and get the current value that might be null since data not yet loaded. And you will not get any further updates from it.

Reply:
And how can operate between flow and livedata?